File: 프로그래머스/lv1/42862. 체육복/체육복.kt
class Solution {
    fun solution(n: Int, lost: IntArray, reserve: IntArray): Int {
        val reserveList = reserve.sorted().toMutableList()
        val lostList = lost.sorted()
        var answer = n - lost.size

        for (i in lostList) {
            if (reserveList.contains(i)) {
                answer++
                reserveList.remove(i)
            } else if (reserveList.contains(i - 1)) {
                answer++
                reserveList.remove(i - 1)
            } else if (reserveList.contains(i + 1)) {
                if (!(lostList.contains(i + 1) && reserveList.contains(i + 1))) {
                    answer++
                    reserveList.remove(i+1)
                }
            }
        }

        return answer
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.swift
func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    var answer = 0
    var lostStudents = lost.sorted()
    var reserveStudents = reserve.sorted()
    
    for i in lostStudents {
        if reserveStudents.contains(i) {
            removeStudent(&lostStudents, i, &reserveStudents, i)
        }
    }
    
    for i in lostStudents {
        if reserveStudents.contains(i-1) {
            removeStudent(&lostStudents, i, &reserveStudents, i-1)
        } else if reserveStudents.contains(i+1) {
            removeStudent(&lostStudents, i, &reserveStudents, i+1)
        }
    }
    
    return n-lostStudents.count
}

func removeStudent(_ lostStudents: inout [Int], _ lostStudent: Int, _ reserveStudents: inout [Int], _ reserveStudent: Int) {
    lostStudents.remove(at: getIndex(lostStudents, lostStudent))
    reserveStudents.remove(at: getIndex(reserveStudents, reserveStudent))
}

func getIndex(_ students: [Int], _ student: Int) -> Int {
    return students.firstIndex(of: student)!
}